assert/assertions.go

```diff
@@ -606,6 +606,22 @@ func Panics(t TestingT, f PanicTestFunc, msgAndArgs ...interface{}) bool {
 	return true
 }
 
+// PanicsWithValue asserts that the code inside the specified PanicTestFunc panics with a specified value.
+//
+//   assert.Panics(t, func(){
+//     GoCrazy()
+//   }, "Panic value", "Calling GoCrazy() should panic with value \"Panic value\"")
+//
+// Returns whether the assertion was successful (true) or not (false).
+func PanicsWithValue(t TestingT, f PanicTestFunc, v interface{}, msgAndArgs ...interface{}) bool {
```

Comment:
Most method's signatures have `expected, actual`. I'm thinking that the order should be `PanicsWithValue(t, "value", func() {})` for consistency.